angelsindustries/control.lua

```diff
@@ -237,8 +232,116 @@ script.on_event({defines.events.on_lua_shortcut,
     local toggled = force.ghost_time_to_live == 0
     force.ghost_time_to_live = toggled and 60 * 60 * 60 * 24 * 7 or 0
 
-    for _, fplayer in pairs(force.players) do
-      fplayer.set_shortcut_toggled(input, toggled)
+    for _, fplayer in pairs(force.players) do fplayer.set_shortcut_toggled(input, toggled) end
+  end
+end)
+
+------------------------------
+-- VEHICLE BURNER EQUIPMENT --
+------------------------------
+script.on_init(function(event)
+  global.vehicle_burners = {}
+  global.current_car = {}
+end)
+
+script.on_configuration_changed(function(event)
+  if global.vehicle_burners == nil then global.vehicle_burners = {} end
+  if global.current_car == nil then global.current_car = {} end
+end)
+
+script.on_event(defines.events.on_gui_opened, function(event)
+  if event.gui_type == defines.gui_type.entity and event.entity ~= nil and event.entity.type == "car" then
+    global.current_car = event.entity
+  end
+end)
+
+script.on_event(defines.events.on_gui_closed, function(event)
+  if event.gui_type == defines.gui_type.entity and event.entity.type == "car" and global.current_car == event.entity then
+    global.current_car = {}
+  end
+end)
+
+script.on_event(defines.events.on_player_placed_equipment, function(event)
+  if event.equipment.name == "angels-burner-generator-vequip" and next(global.current_car) ~= nil then
+    global.vehicle_burners[global.current_car.unit_number] = {
+      vehicle = global.current_car,
+      burner = event.equipment,
+      grid = event.grid
+    }
+  end
+end)
+
+script.on_event(defines.events.on_player_removed_equipment, function(event)
+  if event.equipment == "angels-burner-generator-vequip" and next(global.current_car) ~= nil then
+    global.vehicle_burners[global.current_car.unit_number] = nil
+  end
+end)
+
+script.on_nth_tick(60, function()
```

Comment:
As a follow up on @Pezzawinkle's https://github.com/Arch666Angel/mods/pull/464#pullrequestreview-629001315 comment:
> I'm not sure if we want to increase the tick count a bit higher but im fine with it now.
The thing to consider, is realistically, how many "stacks" of wood do we want to stick in per second?

My suggestion would be more like this:
1) Have a new equipment, named `Burner Refueler` (which could consume a small amount of power)
2) Instead of iterating through each vehicle burner, I would iterate over each refueler, where every refueler:
    - Attempt at filling each burner generating equipment in that equipment grid
    - Stop filling equipment if:
        - All the equipment is full of fuel (can't insert anymore)
        - Has no more fuel to insert (inventory out of fuel)
        - Has inserted more than X amount of KJ worth of energy (this can be coded)
        - Has insufficient energy to insert more fuel (see next bullet)
    - This 'laser' equipment (with 0 range) will have a buffer that we can drain for energy:
        - By base game, steady drain for 'idling'
        - By code, taking a bit of energy for each insert action (measured in KJ/item)
3) Depending on the how much items/s we want to auto insert (this can be a low number), we can tweak how much ticks pass between attempts
4) Player can speed up the filling process by adding more Refuel equipments in a certain equipment grid. Because insertion is measured in items/s, more refuelers are needed for wood compared to higher fuel yielding items.

The code that is here already is the major part of the work, this only requires small tweaks (and addition of a new equipment item) in order to work. This would also resolve incompatibilities with existing vehicle(burners) in existing savegames...

Reply:
From testing it was not moving single items, rather full stacks, but that may have just been doing all 600 transitions in the same tick (in the case of carbon).

Reply:
> From testing it was not moving single items, rather full stacks, but that may have just been doing all 600 transitions in the same tick (in the case of carbon).

it is not. it checks for how much it can insert and then removes that amount and then inserts that amount or max items available which ever one is less.

it is functional currently but i could see adding the fueler equipment as a decent option as then we dont need to care about tracking the generators and only keep track of the fueler equipment which would simplify mod compatibility. i know the og version wasn't setup at all for mod compatibility and wont even work with bobs equipment.